File: Java/src/model/Collections/WrapperStack.java
package model.Collections;

import interfaces.IStack;

import java.util.Stack;

/**
 * Created by Lucian on 11/2/2015.
 */
public class WrapperStack<T> implements IStack<T> {

    /**
     * Storage stack
     */
    private Stack<T> stack;

    /**
     * The constructor
     */
    public WrapperStack(){
        stack = new Stack<>();
    }

    /**
     * Get last element from stack
     * Delete it from stack
     *
     * @return T element
     */
    @Override
    public T pop() {
        return stack.pop();
    }

    /**
     * Push an element in top of the stack
     *
     * @param element The element
     */
    @Override
    public void push(T element) {
        stack.push(element);
    }

    /**
     * Get last element from stack
     *
     * @return T element
     */
    @Override
    public T top() {
        return stack.peek();
    }

    /**
     * Check if the stack is empty
     *
     * @return True if yes, false otherwise
     */
    @Override
    public boolean isEmpty() {
        return stack.isEmpty();
    }

    /**
     * Get the size of the stack
     *
     * @return int size
     */
    @Override
    public int size() {
        return stack.size();
    }

    /** String representation
    ** @return String
    */
    @Override
    public String toString() {
        StringBuilder result = new StringBuilder();
        for(T element : stack){
            result.append(element.toString());
            result.append("\n");
        }
        return result.toString();
    }
}

File: C#/ToyLanguage/ToyLanguage/Model/Collections/WrapperStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguage.Exceptions;
using ToyLanguage.Interfaces;

namespace ToyLanguage.Model.Collections
{
    class WrapperStack<T> : IMyStack<T> where T : IMyStatement
    {
        private Stack<T> stack;
           
        public WrapperStack()
        {
            stack = new Stack<T>();
        }

        public bool isEmpty()
        {
            if(stack.Count == 0)
            {
                return true;
            }
            return false;
        }

        public T pop()
        {
            if(stack.Count > 0)
            {
                return stack.Pop();
            }
            throw new EmptyContainerException();
        }

        public void push(T element)
        {
            stack.Push(element);
        }

        public int size()
        {
            return stack.Count;
        }

        public T top()
        {
            if (stack.Count > 0)
            {
                return stack.Peek();
            }
            throw new EmptyContainerException();
        }

        public string ToString()
        {
            StringBuilder result = new StringBuilder();

            foreach(T element in stack){
                result.Append(element.MyToString());
                result.Append("\n");
               
            }
            return result.ToString();
        }
    }

}

